src/hera/workflows/_unparse.py

```diff
@@ -944,6 +946,10 @@ def unparse(ast_obj):
 
 def roundtrip(source):
     tree = ast.parse(source)
-    if hasattr(ast, "unparse"):
+    # python 3.11 ast.unparse stopped using brackets around tuples used in assignment,
+    # see https://github.com/python/cpython/commit/51cef8be8c77dff522bec6f95d6853031bf19038
+    # Previous versions will unparse with brackets
+    if sys.version_info[1] >= 11 and hasattr(ast, "unparse"):
```

Comment:
Not the current implementation of unparse was taken from python 3.8 compat syntax. Python 3.9+ have added other experssions to the grammar and the parser/unparser may not work as expected. I would prefer if we drop support for 3.8 entirely if possible rather than produce incorrect parse trees.

Reply:
We already see some issues here for eg https://github.com/argoproj-labs/hera/issues/584